nvme.c

```diff
@@ -3957,6 +3957,47 @@ static int list_secondary_ctrl(int argc, char **argv, struct command *cmd, struc
 	return err;
 }
 
+static const char dash[50] = {[0 ... 49] = '='};
+static const char space[50] = {[0 ... 49] = ' '};
+
+static void wait_self_test(struct nvme_dev *dev)
+{
+	static const char spin[] = {'-', '\\', '|', '/' };
+	struct nvme_self_test_log log;
+	int err, i = 0, p = 0;
+
+	printf("Waiting for self test completion...\n");
+	while (true) {
+		if (p == 100) {
```

Comment:
The value is described by the specfication as below but can we check the value as 100% correctly? Also not neccesary to check the value as 0% as completed but not `progress broken`?
```
Current Device Self-Test Completion
If bits 3:0 in the Current Device Self-Test Operation field are cleared to 0h (indicating there is no device self-test operation in progress), then this field is ignored.
```

Reply:
The wait function sleeps every one second so I think whether the value is possible to be changed to zero in the one second period. Anyway I will test it later. By the way about the termination I thought the wait option can be set the timeout threshold also as optionally.

Reply:
I'd rather not exposing the wait timeout on the command line. If you want to be fancy the wait time could be calculated from the previous loop. So if you don't do a lot of progress in 1 second increase the timeout. But this is fancy stuff, I would also like to avoid at this point.

Reply:
Okay I see. Thanks for your comments. Sorry on my local environment the dst command is not supported as below so if any chance I will retry on other environment.
```
[2023-01-13 20:29:06.603] tokunori@tokunori-desktop:~/nvme-cli$ sudo .build/nvme device-self-test /dev/nvme0 -w
[2023-01-13 20:29:07.603] NVMe status: Invalid Field in Command: A reserved coded value or an unsupported value in a defined field(0x2)
[2023-01-13 20:29:07.618] tokunori@tokunori-desktop:~/nvme-cli$ sudo .build/nvme device-self-test /dev/nvme0 -w
[2023-01-13 20:29:09.509] NVMe status: Invalid Field in Command: A reserved coded value or an unsupported value in a defined field(0x2)
[2023-01-13 20:29:09.509] tokunori@tokunori-desktop:~/nvme-cli$
```

Reply:
Thanks for your advice. Yes I could do execute the dst command with the '-s' option as you mentioned as below. And also confirmed the waiting failed as 'progress broken'. (The '-w' option is not checked correctly as you mentoned as always waits without the option.)
```
tokunori@tokunori-desktop:~/nvme-cli$ sudo .build/nvme device-self-test /dev/nvme0 -s 1
Short Device self-test started
Waiting for self test completion...
[===============================================/  ]  95%
progress broken
```
By the way it seems that it is better to be set the '-s' option default valeus as 1 but not as 0.